xline/src/server/auth_server.rs

```diff
@@ -206,18 +207,7 @@ where
     ) -> Result<tonic::Response<AuthUserAddResponse>, tonic::Status> {
         debug!("Receive AuthUserAddRequest {:?}", request);
         let user_add_req = request.get_mut();
-        if user_add_req.name.is_empty() {
-            return Err(tonic::Status::invalid_argument("user name is empty"));
-        }
-        let need_password = user_add_req
-            .options
-            .as_ref()
-            .map_or(true, |o| !o.no_password);
-        if need_password && user_add_req.password.is_empty() {
-            return Err(tonic::Status::invalid_argument(
-                "password is required but not provided",
-            ));
-        }
+        user_add_req.validation()?;
```

Comment:
Can we directly move requests validations that are stateless into xline-client?
This way, we won't need to perform double checks at both the Store and Server.

Reply:
Whether we can do it depends on whether `etcd` validates these stateless requests on its client side. If it doesn't, we have to validate these requests on the server side to be compatible with etcd.

Reply:
The etcdctl will do some validation before it issues a get command. FYI: https://github.com/etcd-io/etcd/blob/main/etcdctl/ctlv3/command/get_command.go#L98